asset/s3_reader/fetcher.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const {
+    Fetcher, getClient
+} = require('@terascope/job-components');
+const { getChunk } = require('@terascope/chunked-file-reader');
+
+class S3Fetcher extends Fetcher {
+    constructor(context, opConfig, executionConfig) {
+        super(context, opConfig, executionConfig);
+        this.client = getClient(context, opConfig, 's3');
+        this._initialized = false;
+        this._shutdown = false;
+    }
+
+    async initialize() {
+        this._initialized = true;
+        return super.initialize();
+    }
+
+    async shutdown() {
+        this._shutdown = true;
+        return super.shutdown();
+    }
+
+    async fetch(slice) {
+        // Coerce the field delimiter if the format is `tsv`
+        if (this.opConfig.format === 'tsv') {
+            this.opConfig.field_delimiter = '\t';
+        }
+        this.reader = (offset, length) => {
```

Comment:
We shouldn't be assigning this function to `this`. 
```suggestion
        const reader = (offset, length) => {
```

Reply:
fixed